sql/core/src/test/resources/sql-tests/results/show-tables.sql.out

```diff
@@ -120,19 +120,9 @@ show_t3
 -- !query
 SHOW TABLE EXTENDED LIKE 'show_t*'
 -- !query schema
-struct<namespace:string,tableName:string,isTemporary:boolean,information:string>
+struct<namespace:string,tableName:string,isTemporary:boolean,information:map<string,string>>
 -- !query output
-	show_t3	true	Table: show_t3
```

Comment:
which is the information column?

Reply:
The output only contains information column.

This is the change of [HiveResult](https://github.com/apache/spark/pull/32563/files#diff-275ec2fd7ca81e5482c959562236c22007d3c2274e456f4e44b78695e37814a4R68-R72).

Example of Hive output :
```
hive> SHOW TABLE EXTENDED LIKE '*';
OK
tableName:spark_32976
owner:yumwang
location:file:/tmp/spark/spark_32976
inputformat:org.apache.hadoop.mapred.TextInputFormat
outputformat:org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
columns:struct columns { i32 id, string name}
partitioned:true
partitionColumns:struct partition_columns { string part}
totalNumberFiles:unknown
totalFileSize:unknown
maxFileSize:unknown
minFileSize:unknown
lastAccessTime:unknown
lastUpdateTime:unknown

tableName:t1
owner:yumwang
location:file:/tmp/hive/t1
inputformat:org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat
outputformat:org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat
columns:struct columns { string id}
partitioned:true
partitionColumns:struct partition_columns { date part}
totalNumberFiles:unknown
totalFileSize:unknown
maxFileSize:unknown
minFileSize:unknown
lastAccessTime:unknown
lastUpdateTime:unknown
```

Reply:
The result doesn't match the command output schema?

Reply:
Yes. Hive only contains one column:
https://github.com/apache/hive/blob/45b48d5fdc3527e56347b297d1a41902d1313220/ql/src/java/org/apache/hadoop/hive/ql/metadata/formatting/TextMetaDataFormatter.java#L248-L263.
The output similar to Spark's `CatalogTable.toLinkedHashMap`.

Reply:
Sorry I'm confused. How can we allow a mismatch between the schema and data?

Reply:
Only get the last column: https://github.com/apache/spark/blob/ca976cdb091c2ba308db37f25968318c3f942ed0/sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala#L69-L72

Reply:
It's only for thrift-server and the SQL shell, but not `sql(...).show`?

Reply:
this is only used in the SQL shell, not by the thrift server. Otherwise, the JDBC `ResultSet` will get wrong for mapping the metadata and column results.

If we turn on the `spark.sql.cli.print.header=true` for SQL shell
1) w/o this PR, the information column matches the first element of the map, then EOL. The rest of the map will print line by line, with wrong/no indentations.
2) w/ this PR, I guess the schema header and the result more inappropriate matched

Reply:
`spark-sql` Before:
```
spark-sql> set spark.sql.legacy.keepCommandOutputSchema=true;
spark.sql.legacy.keepCommandOutputSchema	true
Time taken: 0.012 seconds, Fetched 1 row(s)
spark-sql> SHOW TABLE EXTENDED LIKE '*';
default	test_parquet	false	CatalogTable(
Database: default
Table: test_parquet
Owner: yumwang
Created Time: Mon May 24 11:16:33 CST 2021
Last Access: UNKNOWN
Created By: Spark 3.2.0-SNAPSHOT
Type: MANAGED
Provider: hive
Table Properties: [transient_lastDdlTime=1621826201]
Statistics: 290 bytes
Location: file:/Users/yumwang/tmp/xxxx/spark/spark-warehouse/test_parquet
Serde Library: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
InputFormat: org.apache.hadoop.mapred.TextInputFormat
OutputFormat: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
Storage Properties: [serialization.format=1]
Partition Provider: Catalog
Schema: root
 |-- id: long (nullable = false)
)

Time taken: 0.031 seconds, Fetched 1 row(s)
```

`spark-sql` After:
```
spark-sql> SHOW TABLE EXTENDED LIKE '*';
default	test_parquet	false	{"Created By":"Spark 3.2.0-SNAPSHOT","Created Time":"Mon May 24 11:16:33 CST 2021","Database":"default","InputFormat":"org.apache.hadoop.mapred.TextInputFormat","Last Access":"UNKNOWN","Location":"file:/Users/yumwang/tmp/xxxx/spark/spark-warehouse/test_parquet","OutputFormat":"org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat","Owner":"yumwang","Partition Provider":"Catalog","Provider":"hive","Schema":"root
 |-- id: long (nullable = false)
","Serde Library":"org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe","Statistics":"290 bytes","Storage Properties":"[serialization.format=1]","Table Properties":"[transient_lastDdlTime=1621826201]","Table":"test_parquet","Type":"MANAGED"}
Time taken: 0.043 seconds, Fetched 1 row(s)
```

Reply:
`beeline` before:
```
0: jdbc:hive2://localhost:10000> set spark.sql.legacy.keepCommandOutputSchema=true;
+-------------------------------------------+--------+
|                    key                    | value  |
+-------------------------------------------+--------+
| spark.sql.legacy.keepCommandOutputSchema  | true   |
+-------------------------------------------+--------+
1 row selected (0.055 seconds)
0: jdbc:hive2://localhost:10000> SHOW TABLE EXTENDED LIKE '*';
+-----------+---------------+--------------+----------------------------------------------------+
| database  |   tableName   | isTemporary  |                    information                     |
+-----------+---------------+--------------+----------------------------------------------------+
| default   | test_parquet  | false        | CatalogTable(
Database: default
Table: test_parquet
Owner: yumwang
Created Time: Mon May 24 11:16:33 CST 2021
Last Access: UNKNOWN
Created By: Spark 3.2.0-SNAPSHOT
Type: MANAGED
Provider: hive
Table Properties: [transient_lastDdlTime=1621826201]
Statistics: 290 bytes
Location: file:/Users/yumwang/tmp/xxxx/spark/spark-warehouse/test_parquet
Serde Library: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
InputFormat: org.apache.hadoop.mapred.TextInputFormat
OutputFormat: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
Storage Properties: [serialization.format=1]
Partition Provider: Catalog
Schema: root
 |-- id: long (nullable = false)
)
 |
+-----------+---------------+--------------+----------------------------------------------------+
1 row selected (0.086 seconds)
```
`beeline` after:
```
0: jdbc:hive2://localhost:10000> SHOW TABLE EXTENDED LIKE '*';
+------------+---------------+--------------+----------------------------------------------------+
| namespace  |   tableName   | isTemporary  |                    information                     |
+------------+---------------+--------------+----------------------------------------------------+
| default    | test_parquet  | false        | {"Created By":"Spark 3.2.0-SNAPSHOT","Created Time":"Mon May 24 11:16:33 CST 2021","Database":"default","InputFormat":"org.apache.hadoop.mapred.TextInputFormat","Last Access":"UNKNOWN","Location":"file:/Users/yumwang/tmp/xxxx/spark/spark-warehouse/test_parquet","OutputFormat":"org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat","Owner":"yumwang","Partition Provider":"Catalog","Provider":"hive","Schema":"root
 |-- id: long (nullable = false)
","Serde Library":"org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe","Statistics":"290 bytes","Storage Properties":"[serialization.format=1]","Table Properties":"[transient_lastDdlTime=1621826201]","Table":"test_parquet","Type":"MANAGED"} |
+------------+---------------+--------------+----------------------------------------------------+
1 row selected (0.903 seconds)
```